Clarify that 42NA8 might show sub-commands (#348)
neo4j/docs-status-codes: modules/ROOT/pages/errors/gql-errors/42NA8.adoc

== Status description
error: syntax error or access rule violation - invalid reference in command. Invalid reference in command `{ <<cmd>> }`.

> == Example scenario

Author: We could have an explanation first, but it should be more generic, something along the lines of

This error occurs as a cause of 42001 when trying to run a GRANT, DENY or REVOKE administration command referring to a user, role, database or other entity which does not exist.

> == Example scenario

Reviewer: Then we need to explain all the scenarios. How about something in this line of thought:

```suggestion

== Explanation

This error occurs as the cause of other exceptions, for example, when trying to grant or deny privileges to, or revoke privileges from, a role that does not exist.
Another example is when you try to grant or deny privileges to, or revoke privileges from, a role to perform administration commands on a database that does not exist.

== Example scenario
```

> == Example scenario

Author: One could think so, but that gives just 42001 -> 42N10 as far as I can see. I do not know why, but I know @HannesSandberg is looking at some inconsistences in this area e.g. some errors have a special step for parameters and some have not

> == Example scenario

Reviewer 2: I think it's weird that REVOKE ROLE FROM USER but not GRANT ROLE TO USER have different error sequences so I think we should fix it yes

== Status description
error: syntax error or access rule violation - invalid reference in command. Invalid reference in command `{ <<cmd>> }`.

== Explanation

> This error occurs as the cause of other exceptions, for example, when trying to grant or deny privileges to, or revoke privileges from, a role that does not exist.

Reviewer: ```suggestion

This error occurs as the cause of other exceptions, for example, when you try to:

* Grant or deny privileges to, or revoke privileges from, a role that does not exist.
```

> Another example is when you try to grant or deny privileges to, or revoke privileges from, a role to perform administration commands on a database that does not exist.

Reviewer: ```suggestion
* Grant or deny privileges to, or revoke privileges from, a role to perform administration commands on a database that does not exist.
```
